src/pois.rs

```diff
@@ -144,10 +145,35 @@ pub async fn index_pois(
             )
         })
         .filter_map(|poi_res| futures::future::ready(poi_res.ok()))
-        .collect()
+        .map(|p| (p.id.clone(), p.clone()))
+        .collect::<HashMap<String, Poi>>()
         .await;
 
-    import_pois(client, settings.container, futures::stream::iter(pois)).await
+    // Add children
+    info!("building pois hierarchy");
+    for (parent_id, parent) in pois {
+        if parent.children.is_empty() {
+            continue;
+        }
+        let children: Vec<Poi> = parent
+            .children
+            .iter()
+            .filter_map(|ch| {
+                let place = match places.get(&format!("{}{}", "poi:", *ch)) {
+                    Some(p) => Some(p.clone()),
+                    _ => {
+                        warn!("Child not found for {}", ch);
+                        None
+                    }
+                };
+                place
+            })
+            .collect();
+        let place = places.get_mut(&format!("{}{}", "poi:", parent_id)).unwrap();
+        place.children = children;
```

Comment:
I think you can avoid making a lookup here 
```suggestion
        parent.children = children;
```

If you loop using : 
```
for (parent_id, parent) in pois.iter_mut() {
```

Reply:
Hum, no, forget about that. I mixed up `pois` and `places`

Reply:
Other comment then : are you sure your `unwrap() `will never panic ?

- if yes, please add a comment explaining why
- if no, you should handle the failure case in some way that does not crash the program